packages/clarity/contracts/taral/taral-exporter-v1.clar

```diff
@@ -38,35 +54,48 @@
         (asserts! (> (len exporter-category) u0) ERR-GENERIC) 
-            (let ((exporter-id (unwrap! (get-or-create-exporter-id exporter) ERR-GENERIC)))
-            (unwrap! (contract-call? .exporter-storage add-exporter-profile exporter-id exporter-name exporter-category) exporter-storage-error)
-            (print {action: "register", exporter: exporter, exporter-name: exporter-name, exporter-category: exporter-category })
-            (ok true)
+        ;; check that the hash is not empty
+        (asserts! (> (len hash) u0) ERR_EMPTY_HASH)
+            
+        (let ((exporter-id (unwrap! (get-or-create-exporter-id exporter) ERR-GENERIC)))
+        (unwrap! (contract-call? .exporter-storage add-exporter-profile exporter-id exporter-name hash exporter-category) exporter-storage-error)
+        (print {action: "register", exporter: exporter, exporter-name: exporter-name, exporter-category: exporter-category })
+        (ok true)
         )
     ) 
 )
 
 ;; @Desc appends order to exporter and updates exporter profile
 ;; @params new-order-id : ID of new order
 ;; @Params exporter : Principal of exporter
-(define-public (append-order (new-order-id uint) (exporter principal))
-    (begin
+(define-public (append-order 
+    (new-order-id uint) 
+    (exporter principal) 
+    (hash (buff 256)) 
+)
+    (let (
+        (exporter-id (unwrap! (contract-call? .exporter-storage get-exporter-by-principal exporter ) ERR-EXPORTER-NOT-REGISTERED))
+        (current-exporter (unwrap! (contract-call? .exporter-storage get-exporter-profile exporter) exporter-storage-error))
+        (new-id (contract-call? .exporter-storage get-orders-next-avail-id current-exporter))
+        )
         (asserts! (not (is-none (contract-call? .exporter-storage get-exporter-by-principal exporter))) ERR-EXPORTER-NOT-REGISTERED)
-            (let (
-                (exporter-id (unwrap! (contract-call? .exporter-storage get-exporter-by-principal exporter ) ERR-GENERIC))
-                (current-exporter (unwrap! (contract-call? .exporter-storage get-exporter-profile exporter) exporter-storage-error))
-                (new-id (contract-call? .exporter-storage get-orders-next-avail-id current-exporter))
-                )
-            (unwrap! (contract-call? .exporter-storage update-exporter-profile {exporter-id: exporter-id} (merge current-exporter { orders-next-avail-id: (+ u1 new-id)})) exporter-storage-error)
-            (unwrap! (contract-call? .exporter-storage add-order new-id exporter-id new-order-id) exporter-storage-error)
-            (print {action: "append-order", exporter: exporter, new-order-id: new-order-id  })
-            (ok true)
-            )
+        ;; check that the hash is not empty
+        (asserts! (> (len hash) u0) ERR_EMPTY_HASH)
+        (unwrap! (contract-call? .exporter-storage update-exporter-profile {exporter-id: exporter-id} (merge current-exporter { orders-next-avail-id: (+ u1 new-id)})) exporter-storage-error)
```

Comment:
@dorucioclea do u think we should merge the hash property here along with incrementing `orders-next-avail-id` cause we are updating the exporter profile and that means change in data right. 

```suggestion
        (unwrap! (contract-call? .exporter-storage update-exporter-profile {exporter-id: exporter-id} (merge current-exporter { orders-next-avail-id: (+ u1 new-id), hash: hash })) exporter-storage-error)
```